qt/aqt/webview.py

```diff
@@ -448,15 +448,31 @@ def standard_css(self) -> str:
             )
 
         zoom = self.app_zoom_factor()
-
         return f"""
 body {{ zoom: {zoom}; background-color: var(--canvas); }}
 html {{ {font} }}
 {button_style}
-:root {{ --canvas: {colors.CANVAS[0]} }}
-:root[class*=night-mode] {{ --canvas: {colors.CANVAS[1]} }}
+:root {{ --canvas: {colors.CANVAS["light"]} }}
+:root[class*=night-mode] {{ --canvas: {colors.CANVAS["dark"]} }}
 """
 
+    def css_vars(self) -> str:
+        br = "\n  "
+        return f"""
+:root {{
+  {br.join([f"/* {color['comment']} */{br}--{name.lower().replace('_', '-')}: {color['light']};" for name, color in colors.__dict__.items() if name != "__builtins__" and isinstance(color, dict)])}
+}}
+:root[class*=night-mode] {{
+  {br.join([f"/* {color['comment']} */{br}--{name.lower().replace('_', '-')}: {color['dark']};" for name, color in colors.__dict__.items() if name != "__builtins__" and isinstance(color, dict)])}
+}}
+:root {{
+  {br.join([f"/* {prop['comment']} */{br}--{name.lower().replace('_', '-')}: {prop['light']};" for name, prop in props.__dict__.items() if name != "__builtins__" and isinstance(prop, dict)])}
+}}
+:root[class*=night-mode] {{
+  {br.join([f"/* {prop['comment']} */{br}--{name.lower().replace('_', '-')}: {prop['dark']};" for name, prop in props.__dict__.items() if name != "__builtins__" and isinstance(prop, dict)])}
+}}
+        """
```

Comment:
Yesterday I had this shower-thought that we could go full-circle with the Sass variables and actually add them to our webviews from the Python side.

At first it sounds kind of dumb, but having a single source of truth will make it much easier for us and add-on authors to apply new themes.

Reply:
Agreed that a single source of truth is good, but it's best if that is the Rust or TypeScript, which is shared by all clients. Any complexity we push into the Python code means that the two mobile clients need up having to reimplement it, which wastes developer resources.  Could these comments not be included via our JS/sass tooling instead?

Likewise for the rgba solution - you've added extra parsing code into Python that the mobile clients will now need to mirror; couldn't we instead limit ourselves to hex colors of the form #aabbcc or #aabbccdd in our color definitions, to keep the parsing on the clients simpler?

Reply:
This is the only thing I'd like clarified before merging. It looks like the comments are included in .bazel/bin/sass/_root_vars.css. Is this Python code serving some other purpose? And if so, can we not accomplish this in our scss->css transform step instead?

Reply:
> Could these comments not be included via our JS/sass tooling instead?

Yes, I've already done that before I went for the Python route. The only reason for doing this is future theme support. If our whole UI consisted of web components, there'd be no need for this step - but we also need to modify Qt widgets via Python with the same variables.

Reply:
If there's a way to do this via Rust/Typescript, I'm all ears.

Reply:
I fear I'm not yet understanding the problem/goal here.  Is your thinking here that the theme author will mutate the values in colors.py, and then we'll automatically regenerate vars.css based on those? Why can't we go in the opposite direction instead, where the theme author provides an alternative vars.css which we then extract the colors from, by including/copying extract_sass_vars.py?

Reply:
(if we want to make it easy to tweak only a few values in the scss file, another option would be to make the source of truth a proto message with defaults provided by the backend, and provide a backend method to generate vars.css from a provided collection of colors)

Reply:
> Is your thinking here that the theme author will mutate the values in colors.py, and then we'll automatically regenerate vars.css based on those?

I thought we'd apply theme changes via `ThemeManager` and `ProfileManager`, so I'd have added a colors dictionary in `ThemeManager`, which would then be modified on theme change.

> Why can't we go in the opposite direction instead, where the theme author provides an alternative vars.css which we then extract the colors from, by including/copying extract_sass_vars.py?

That was my plan for our own theme support. Users would provide a CSS or JSON file from which we extract the colors. Add-on authors (e.g. ReColor, Anki Redesign) on the other hand might want to control things via Python instead - at least that's what I was thinking.

>  make the source of truth a proto message with defaults provided by the backend

That sounds a bit cumbersome for us developers. We'd lose all Sass features such as `color.scale`or `.adjust`. I also wanted to auto-generate a `gradient-middle` variable from the `gradient-start` and `gradient-end` values via Sass functions in the future.

I'm not familiar enough with the backend to see this problem from your perspective, yet.d

Reply:
I've reverted the changes, because I don't want to hold back this update any longer than necessary. We should continue this discussion in a proper theming PR in the future :)

Reply:
Agreed on continuing this later :-)